### PR TITLE
Remove auto approvals

### PR DESCRIPTION
**Pull request by jofeinstein**
Remove Project Auto-Approvals

- Removed all text about auto-approvals in templates
- removed `auto_approve_project_join_requests` from `project/utils.py`
- removed auto-approval paths from  `project/urls.py`
- removed `ProjectAutoApproveJoinRequestsView`
- altered views to remove auto approval code
- Removed auto-approval fields in forms `ProjectReviewUserJoinForm` and `ProjectUpdateForm`
- removed auto-approval fields in Project model
	- migration file `project/migrations/0015` reflects these changes

Commands
- added command `project/management/commands/pending_join_request_reminder.py`
	- sends email to PIs if there are join requests older than 4 days
	- can change variable `TIME_DELTA` to alter the email sending interval
	- to run, execute `python manage.py pending_join_request_reminder`
- added permission to `create_staff_group.py`
 	- will need to be rerun for the staff group to reflect changes

Tests
- added tests to ensure project join requests, project join request approvals/denials, and project updating are still functional 
	-  `coldfront.core.project.tests.test_views.test_auto_approval_removal`
- added tests for new pending_join_request_reminder command
	- `coldfront.core.project.tests.test_commands.test_pending_auto_request_reminder`

### CODE REVIEW

**Comment by matthew-li** [comment]
@jofeinstein

From a conversation with @wfeinstein, we should:
1. Make it easy for a user to see their pending join requests. Right now, the only way to see these is Project -> Projects -> Join a Project, which is not obvious. Adding a link to the home page directing them there seems good enough to me.
    - One option would be to display an alert somewhere in the "My BRC Cluster Projects" section: "You have K requests to join projects that are pending manager approval." (or something similar) if and only if `K > 0`.
2. Add text to encourage users to contact the PI/managers in that join view.
3. Provide staff and superusers with a view to see pending join requests across all projects, since currently they have to look at individual project pages.

**Review by matthew-li — CHANGES_REQUESTED** [review]
Looks reasonable overall, thanks for addressing this.

I've left some comments regarding how/to whom emails get sent.

File: coldfront/core/project/management/commands/pending_join_request_reminder.py
Comment (jofeinstein) [reply coldfront/core/project/management/commands/pending_join_request_reminder.py]: So we only want to send emails to PIs that have `enable_notifcations=True`?

File: coldfront/core/project/management/commands/pending_join_request_reminder.py
Comment (jofeinstein) [reply coldfront/core/project/management/commands/pending_join_request_reminder.py]: Thats true, but now we are just using this query set to find all projects and users with pending join requests. Having a previously denied request and a new pending request should not affect this.

Comment (jofeinstein) [reply coldfront/core/project/management/commands/pending_join_request_reminder.py]: I also changed altered some query sets below in this [commit](https://github.com/ucb-rit/coldfront/pull/302/commits/8f0c2044090e2a89df302824978514432fdf79d0) so that only the latest request is shown in an email if there were previous denied requests.

File: coldfront/core/project/management/commands/pending_join_request_reminder.py
Comment (matthew-li) [reply coldfront/core/project/management/commands/pending_join_request_reminder.py]: Yes.

**Comment by matthew-li** [comment]
As an unauthenticated user, I'm getting an `'AnonymousUser' object is not iterable` error on the home page.

We should only try to fetch join requests for authenticated users.

**Comment by jofeinstein** [comment]
Got it

**Comment by matthew-li** [comment]
@jofeinstein Thanks for going through the back-and-forth. I added a few email-related tweaks.

I think this is ready to merge after:
(a) We get confirmation on when reminders should start getting sent, and
(b) A view is added for staff/administrators to see join requests across all projects (see one of the comments above).

**Comment by jofeinstein** [comment]
@matthew-li 

The new view for viewing pending join requests is `ProjectJoinRequestListView`. Tests for the view are located at `cold front/core/project/tests/test_views/test_project_join_request_list_view.py`

**Comment by jofeinstein** [comment]
I also added a new permission to the staff group for the above view.

**Review by matthew-li — APPROVED** [review]
I renamed the migration to be 0016 to avoid conflicting with the newly-added 0015 from your other PR.

I also updated how dates are checked in `TestProjectJoinRequestListView` to use the proper display time zone (`America/Los_Angeles`).

Thanks!

**Pull request merged**
